### ISSUE TITLE
(3P) FSE: a11y feedback

### ISSUE THREAD

**Issue by getdave** [Automattic/wp-calypso#33405]
Gain feedback on and test the a11y of the feature.

**Comment by getdave** [comment]
> An <input> element with a type="button" declaration and a valid value attribute does not need a label associated with it. Doing so may actually interfere with how assistive technology parses the button input. The same applies for the `<button>` element.

https://developer.mozilla.org/en-US/docs/Web/HTML/Element/label

**Comment by getdave** [comment]
Waiting on feedback `p4vt7e-4Q-p2`

**Comment by davidakennedy** [comment]
👋 I'm not done with all my testing, but wanted to share stuff as I had it. I tested Blog Post Listing blog on the front end. The markup could be improved:

*Post dates*

Please add a "Posted on", hidden via CSS before each post date. See the pattern used in Underscores: https://github.com/Automattic/_s/blob/master/inc/template-tags.php#L29

Why? Adding this gives me context to screen reader users because otherwise, it's just a random date on the page.

Generally, we mark up the dates with the `<time>` element. That's not really related to accessibility, just better HTML5 markup. See Underscores: https://github.com/Automattic/_s/blob/master/inc/template-tags.php#L17 It's also a good idea to make the date a link to the post. Why? In case there isn't a post title a user still has a way to get to the single view of the post.

*Edit link*

Please add the post title after "Edit" hidden via CSS.

Why? Adding this gives me context to screen reader users because otherwise, it's just a bunch of edit links on the page that all say the same thing, but go to different places.

*Read more*

Similar to the edit link, please add a post title after the "Read more" links, hidden via CSS. Something like Read more `<span class="visually-hidden">of Post Title</span>`.

Why? Adding this gives me context to screen reader users because otherwise, it's just a bunch of read more links on the page that all say the same thing, but go to different places.

**Comment by obenland** [comment]
> I'm not done with all my testing, but wanted to share stuff as I had it. I tested Blog Post Listing blog on the front end. 

Tracked in #33538

**Comment by davidakennedy** [comment]
I'm back with a bit more feedback. 😄 Overall, this is a really solid experience from an accessibility experience. I can tell a lot of thought was put into the markup.

Here are some things I have questions on or think could be improved:

1. Technically, `<div>` elements are not allowed to be child elements of `<button>` elements. This occurs in the buttons to select the templates. It's invalid HTML. Those would work as `<span>` elements. Note: This doesn't seem to cause any accessibility issues, just pointing it out because it might cause some or lead to weird display issues.

2. I'm not sure what this label belongs to. It seems to be orphaned: `<label class="components-base-control__label" for="inspector-radio-control-1">Template</label>`. It doesn't make sense in any context that I can see.

3. The alternative text for the images can be improved. I think we can stick to two sentences here.

Blank: Start with a blank template. No content will be added.
Menu: Start with a template that creates a restaurant menu. Contains a large full-width image at the top, followed by a three-column set of images and then several alternating sets of images and text.
Contact us: Start with a template that creates a contact page. Contains a large full-width image at the top, followed by text, a contact form and a three-column set of images for social media information.
Home: Start with a template that creates a home page. Contains a large full-width image at the top, followed by several alternating sets of images and text, text for an address, a contact form and copyright text.

4. I see a `<fieldset>` element but no `<legend>` for it. The "Select Page Template" text seems best for it.

**Comment by jeryj** [comment]
Following up here:
> Technically, <div> elements are not allowed to be child elements of <button> elements.

I changed the `<div>` to a `<span>`. Easy enough :) https://github.com/Automattic/wp-calypso/pull/38916

> I'm not sure what this label belongs to. It seems to be orphaned: <label class="components-base-control__label" for="inspector-radio-control-1">Template</label>. It doesn't make sense in any context that I can see.

This seems to be gone now.

> The alternative text for the images can be improved. 

It appears that the `alt` text is empty for all images now, so it's essentially just considered presentational. However, the images are supposed to closely mirror the output, so it seems like these should have a descriptive `alt`?

> I see a `<fieldset>` element but no `<legend>` for it. The "Select Page Template" text seems best for it.

Each `<fieldset>` section now has a `<legend>`. The two are are currently "Recommend Layouts" and "Homepage Layouts". I think these are planning to be changed, but they at least _do_ have a `<legend>` now :)

**Comment by frontdevde** [comment]
Issue triage 27.02.2020: First pass at addressing feedback was done. If we need another pass, let's open a new issue referencing this one. Closing stale issue for now.
